### ISSUE TITLE
R&S Back to Top Accessibility Update

### ISSUE THREAD

**Issue by brianalloyd** [department-of-veterans-affairs/va.gov-team#16532]
1. We have a sticky button that is located in the page source just above the footer. This works for sighted users, and is okay as-is, except rather than placing it just above the footer it should be the last item in the parent container (see sketch below). _One caveat, @jenstrickland connecting with @1Copenut — to learn if we should use aria-live to announce the button for assistive tech._

2. An option for non-sighted users is needed. We may accomplish this with a `<button>` that is screen reader only, so it announces to assistive tech users in the page flow. This should be positioned above every `h2` except the first one. The last one is the one that sighted users see. This behavior should not change any visual design.

```
<Parent container>

  <h2>Heading 2</h2>
  <p> Section content, may be multiple paragraphs</p>
  <button>Back to top</button>

  <h2>Heading 2</h2>
  <p> Section content, may be multiple paragraphs</p>
  <button>Back to top</button>

  <h2>Heading 2</h2>
  <p> Section content, may be multiple paragraphs</p>
  <button>Back to top</button>

  <h2>Heading 2</h2>
  <p> Section content, may be multiple paragraphs</p>

  <button>Back to top</button><!-- This is the one sighted users see -->
</Parent container>

<Complementary info>
  <p>Content</p>
</Complementary info>

<footer>
</footer>
```
##Planning Considerations
- Reference Ticket [#16192](https://github.com/department-of-veterans-affairs/va.gov-team/issues/16192)
## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] The data returned matches API response (for given user or scenario)
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

**Comment by youngfreezyVA** [comment]
hi @jenstrickland hope you had a good long weekend. question for you... am I to read this as, either implement option 1 (pending your research) _or_ option 2?
or, implement option 1 _and_ option 2?
Is this ticket ready to begin development or do I need to wait for you to connect with @1Copenut?
Many thanks,
Fareez

**Comment by jenstrickland** [comment]
@youngfreezyVA — The recommendation is:

1. Move the current visible sticky button to the bottom of the parent container, so it's the last one, and make that the persistent one for sighted users, and the last one for non-sighted users.

1. Add additional screen reader button above every `h2` except the first and last, and use the screen reader only class to hide it for sighted users. 

Please let me know if anything is unclear. 

[I never heard from Trevor about `aria-live` so I checked with my Gov.uk Design System and accessibility contacts about aria-live, thereby made the decision not to use `aria-live`.]

**Comment by 1Copenut** [comment]
Is this component used outside your team at the moment? I've got another audit I'm running for the App Directory team that uses a button for Back to Top. JAWS doesn't set focus correctly in the audit I'm working on, and I want to make sure I'm ticketing properly.

**Comment by jenstrickland** [comment]
@1Copenut Do you have a link to that work? That's the kind of challenge I worry about, and why I recommended the above.

**Comment by 1Copenut** [comment]
@jenstrickland Yep: https://www.va.gov/resources/connected-apps-faqs/ The button appears if you start scrolling down the screen a bit.

**Comment by jenstrickland** [comment]
@1Copenut The button receives focus as the last item before the footer, which is similar to the R&S implementation. 

This means there's no opportunity for assistive tech to access the button. As noted above, we're adding a screen reader only button above each h2 except the first and last. Previously we considered using aria to offer the button, but after checking with other a11y folk opted out due to reports of erratic behavior in screen readers. Does that seem sound to you, or would you recommend we use aria to be consistent with other things on VA.gov?

**Comment by 1Copenut** [comment]
@jenstrickland I should have couched my comment differently. The button will receive focus, but when I press the button or click it, the target at the top of the page doesn't maintain focus correctly. When I press `DOWN_ARROW` in JAWS again, it moves to the first heading in the footer, because the button still has virtual cursor focus.

**Comment by jenstrickland** [comment]
@1Copenut Oof. That sounds like the same focus management issue we have for the "On this page" jump links. Do we need to dynamically/programmatically set a `tabindex="-1"` on the top of the page id, or might we code that tabindex generally rather than on the fly?
